core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -431,11 +431,11 @@ public void onLeadershipChange(Set<Partition> partitionsBecomeLeader,
 
             leaderPartitions.forEach(this::cacheTopicPartitionIds);
             followerPartitions.forEach(this::cacheTopicPartitionIds);
-            followerPartitions.forEach(this::removeRemoteTopicPartitionMetrics);
 
             remoteLogMetadataManager.onPartitionLeadershipChanges(leaderPartitions, followerPartitions);
             followerPartitions.forEach(topicIdPartition ->
                     doHandleLeaderOrFollowerPartitions(topicIdPartition, RLMTask::convertToFollower));
+            followerPartitions.forEach(this::removeRemoteTopicPartitionMetrics);
```

Comment:
It is good to add a comment here so that it helps readers on why the metrics are removed here.